# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/esden/wvstreams/openssl-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./crypto/wvx509.cc.orig	2004-11-18 15:15:05.000000000 +0100
+++ ./crypto/wvx509.cc	2005-07-19 14:35:09.000000000 +0200
@@ -1162,12 +1162,19 @@
 		else
 		{
 		    void *ext_data = NULL;
+#if OPENSSL_VERSION_NUMBER >= 0x0090800fL
+                   const unsigned char **ext_value_data;
+                   ext_value_data = (const_cast<const unsigned char **>
+                                     (&ext->value->data));
+#else
+                   unsigned char **ext_value_data = &ext->value->data;
+#endif
 		    if (method->it) 
-			ext_data = ASN1_item_d2i(NULL, &ext->value->data, 
+                       ext_data = ASN1_item_d2i(NULL, ext_value_data,
 						ext->value->length, 
 						ASN1_ITEM_ptr(method->it));
 		    else
-			ext_data = method->d2i(NULL, &ext->value->data, 
+                       ext_data = method->d2i(NULL, ext_value_data,
 					      ext->value->length);
 		    
 		    if (method->i2s)
--- ./crypto/wvtripledes.cc.orig	2004-11-18 15:15:05.000000000 +0100
+++ ./crypto/wvtripledes.cc	2005-07-19 14:35:09.000000000 +0200
@@ -102,7 +102,8 @@
 	// ECB works 64bits at a time
 	while (len >= 8)
 	{
-#if OPENSSL_VERSION_NUMBER >= 0x0090705FL
+#if OPENSSL_VERSION_NUMBER >= 0x0090705fL \
+    && OPENSSL_VERSION_NUMBER < 0x0090800fL
 	    DES_ecb3_encrypt(data, crypt,
 			     &deskey1, &deskey2, &deskey3,
 			     mode == ECBEncrypt ? DES_ENCRYPT : DES_DECRYPT);
